packages/connect/src/auth.ts

```diff
@@ -33,6 +33,26 @@ export interface AuthOptions {
   };
 }
 
+/**
+ * Firefox and mobile should not use a 'popup' type of window.
+ */
+export const shouldUsePopup = () => {
+  const isMobile = !!('onTouchStart' in document.documentElement);
+
+  // const isFirefox = navigator.userAgent.toLowerCase().includes('firefox');
+  // return !isMobile && !isFirefox;
+
+  return !isMobile;
+};
+
+export const getOrCreateUserSession = (userSession?: UserSession): UserSession => {
+  if (!userSession) {
+    const appConfig = new AppConfig(['store_write'], document.location.href);
+    userSession = new UserSession({ appConfig });
+  }
+  return userSession;
+};
```

Comment:
I'm wondering if it makes sense to allow folks to pass in their app config scopes, so that we can remove the need for them to generate a usersession, while still being able to define scopes?

Reply:
I like the idea. My intention was only to use this internally, since I had to use the same logic twice. I would definitely be open to your idea (probably as a separate function). We're getting into "get rid of UserSession" territory 😄